File: SDKs/github.com/ghes-3.0/1.1.4/python/src/sdk/models/operations/repos_delete_access_restrictions.py
from dataclasses import dataclass, field



@dataclass
class ReposDeleteAccessRestrictionsPathParams:
    branch: str = field(default=None, metadata={'path_param': { 'field_name': 'branch', 'style': 'simple', 'explode': False }})
    owner: str = field(default=None, metadata={'path_param': { 'field_name': 'owner', 'style': 'simple', 'explode': False }})
    repo: str = field(default=None, metadata={'path_param': { 'field_name': 'repo', 'style': 'simple', 'explode': False }})
    

@dataclass
class ReposDeleteAccessRestrictionsRequest:
    path_params: ReposDeleteAccessRestrictionsPathParams = field(default=None)
    

@dataclass
class ReposDeleteAccessRestrictionsResponse:
    content_type: str = field(default=None)
    status_code: int = field(default=None)
    

File: SDKs/github.com/ghes-2.22/1.1.4/typescript/src/sdk/models/operations/reposdeleteaccessrestrictions.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";


export class ReposDeleteAccessRestrictionsPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=branch" })
  branch: string;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=owner" })
  owner: string;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=repo" })
  repo: string;
}


export class ReposDeleteAccessRestrictionsRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: ReposDeleteAccessRestrictionsPathParams;
}


export class ReposDeleteAccessRestrictionsResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;
}
